File: Punto1/estudiante.cpp
#include <iostream>
#include <string>

using namespace std;

// Definición de la estructura Estudiante
struct Estudiante {
  string nombre;
  int edad;
  float promedio;
};

int main() {
  // Ejemplo de uso
  Estudiante est1;
  est1.nombre = "Juan Perez";
  est1.edad = 20;
  est1.promedio = 4.5;

  cout << "Nombre: " << est1.nombre << endl;
  cout << "Edad: " << est1.edad << endl;
  cout << "Promedio: " << est1.promedio << endl;

  return 0;
}

File: Punto1/estudiante.py
from dataclasses import dataclass

@dataclass
class Estudiante:
    """
    Representa un estudiante con nombre, edad y promedio.
    Utilizamos dataclass como equivalente moderno a un record.
    """
    nombre: str
    edad: int
    promedio: float

# Ejemplo de uso
if __name__ == "__main__":
    est1 = Estudiante(nombre="Juan Perez", edad=20, promedio=4.5)
    print(f"Nombre: {est1.nombre}")
    print(f"Edad: {est1.edad}")
    print(f"Promedio: {est1.promedio}")
